### DISCUSSION TITLE
ROCKET EXAMPLS TLS QUESTION :Warning: secrets enabled without a stable `secret_key`    >> disable `secrets` feature or configure a `secret_key`    >> this becomes an error in non-debug profiles

### DISCUSSION THREAD

**Discussion by Augenstern0320sh** [rwf2/Rocket#2737] [Questions]
cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.21s
     Running `/root/mtls/Rocket-0.5/examples/target/debug/tls`
🔧 Configured for debug.
   >> address: 127.0.0.1
   >> port: 8000
   >> workers: 2
   >> max blocking threads: 512
   >> ident: Rocket
   >> IP header: X-Real-IP
   >> limits: bytes = 8KiB, data-form = 2MiB, file = 1MiB, form = 32KiB, json = 1MiB, msgpack = 1MiB, string = 8KiB
   >> temp dir: /tmp
   >> http/2: true
   >> keep-alive: 5s
   >> tls: enabled w/mtls
   >> shutdown: ctrlc = true, force = true, signals = [SIGTERM], grace = 2s, mercy = 3s
   >> log level: normal
   >> cli colors: true
   >> secret key: [generated]
Warning: secrets enabled without a stable `secret_key`
   >> disable `secrets` feature or configure a `secret_key`
   >> this becomes an error in non-debug profiles
📬 Routes:
   >> (hello) GET / [2]
   >> (mutual) GET /
📡 Fairings:
   >> Shield (liftoff, response, singleton)
   >> HTTP -> HTTPS Redirector (liftoff)
🛡️ Shield:
   >> X-Content-Type-Options: nosniff
   >> X-Frame-Options: SAMEORIGIN
   >> Permissions-Policy: interest-cohort=()
🚀 Rocket has launched from https://127.0.0.1:8000
🚀 Rocket has launched from http://127.0.0.1:3000

Then,Access https://127.0.0.1:8000/ and http://127.0.0.1:3000 Display inaccessible.
How to deal with it? Thanks a lot!

**Comment by SergioBenitez** [comment]
What is the question exactly? Do you want to silence the warning? If so, do as the warning suggests: set a `secret_key`.

**Reply by Augenstern0320sh** [reply]
Thanks for your reply, may I ask how to set a secret_key in the programe examples/tls,?

**Reply by wezm** [reply]
You need to set it in your `Rocket.toml` or via `ROCKET_SECRET_KEY` environment variable. The value can be generated with a command like `openssl rand -base64 32`

More info here: https://rocket.rs/v0.5/guide/requests/#secret-key and here https://rocket.rs/v0.5/guide/configuration/#overview
